code/exercises/affine-cipher/example.scm

```diff
@@ -0,0 +1,74 @@
+(import (rnrs))
+
+
+(define (encode key text)
+  ((get-translator key) (string-downcase text) 'encode))
+
+(define (decode key text)
+  ((get-translator key) text 'decode))
+
+;;; Private
+
+(define (atbash-format lst)
+  (let* ((chunks (windows lst 5))
+         (l (cons (car chunks)
+                  (map (lambda (chars) (cons #\space chars))
+                       (cdr chunks)))))
+    (list->string (fold-left append '() l))))
+
+(define (get-translator key)
+  (let ((a (car key))
+        (b (cdr key)))
+  (when (not (coprime? a 26))
+    (error 'affine-cipher "a and m must be coprime" a))
+  (translator-factory a b)))
+
+;; Work Horse - return a translator tuned to a specific affine key
+(define (translator-factory a b)
+  (let* (;; generate a char pair from an alphabetic index
+         (index->pair
+          (lambda (idx)            ;; (a * x + b) % m
+            (let ((cipher-idx (modulo (+ (* a idx) b) 26)))
+              `(,(idx->char idx) . ,(idx->char cipher-idx)))))
+         ;; build an alist mapping the alphabet to the 'cipherbet'
+         (encode-map (map index->pair (iota 26)))
+         ;; build an alist mapping the 'cipherbet' to the alphabet
+         (decode-map (map riap encode-map))
+         ;; en/decrypt text and return a char list
+         (translate
+          (lambda (text tr-map)
+            (filter
+             ;; identity - to remove #f from the list
+             (lambda (x) x)
+             (map (lambda (c)
+                    (cond ((assoc c tr-map) => cdr)
+                          ((char-numeric? c) c)
+                          (else #f)))
+                  (string->list text))))))
```

Comment:
To keep the `translator`factory` function from exceeding 21 lines I could easily pull this `translate` function out, since it doesn't rely on the surrounding continuation for its data.

I don't think it is of utmost importance to do so, but the style guide would suggest that I do this.